### ISSUE TITLE
Cop to detect sort/sort_by without use of return value

### ISSUE THREAD

**Issue by mmb** [bbatsov/rubocop#4727]
Detect the use of `sort` and `sort_by` without using the return value. This probably means the user expected the change to be in place and they should be using `sort!`

```ruby
x = [1, 2, 3]
x.sort
x
```

**Comment by pocke** [comment]
The cop should detect `select`, `reject`, `sub`, `gsub` etc also.

**Comment by Drenmi** [comment]
Yes. This applies to a lot of methods. It also will likely have a few false positives, but I think that's okay, as it can encourage people who implement standard library-like objects to conform to the same interface. 🙂
